htetl/tasks/main.py

```diff
@@ -0,0 +1,150 @@
+import luigi
+import logging
+import pandas as pd
+import networkx as nx
+import string
+import htetl.extract.emails
+import htetl.get_data as get_data
+import htetl.reverse_url as reverse_url
+import htetl.tasks.phones
+import htetl.util as util
+
+
+logger = logging.getLogger('luigi-interface')
+logger.setLevel(logging.DEBUG)
+
+
+class MakeGraph(luigi.Task):
+    """  reversing url task """
+
+    NEW_DATA_EXTRACTOR = luigi.BoolParameter(
+        config_path=dict(section='htetl-flags', name='new_data_extractor'),
+        default=False
+    )
+
+    def output(self):
+        out_path = 'data/graph.csv'
+        return luigi.LocalTarget(out_path)
+
+    def requires(self):
+        if self.NEW_DATA_EXTRACTOR:
+            return {
+                'email': htetl.extract.emails.ParseEmails(),
+                'phone': htetl.tasks.phones.ParsePhones(),
+                # TODO: This needs to be adapted to the new Page data
+                # extraction methods like ParseEmail and ParsePhones
+                'oid': get_data.RawPosterData(),
+            }
+        else:
+            return [get_data.RawPhoneData(),
+                    get_data.RawEmailData(),
+                    get_data.RawPosterData()]
+
+    def run(self):
+        """ data is small enough to use pandas for the processing """
+
+        out = []
+
+        # TODO: When htetl-flags:new_data_extractor is removed, remove the
+        # flatten call and clean up the for loop
+        for in_target in luigi.task.flatten(self.input()):
+            with in_target.open('r') as f:
+                data = pd.read_csv(f)
+
+            for i, (k, v) in enumerate(data.groupby(data.columns[-1])):
+                values = v.values.tolist()
+                page_ids = [x[0] for x in values]
+                offset_page_ids = page_ids[1:]
+                if len(page_ids) == 1:
+                    offset_page_ids = page_ids
+                else:
+                    offset_page_ids.append(page_ids[0])
+                out.append([
+                    (a, b)
+                    for a, b in zip(page_ids, offset_page_ids)
+                ])
+
```

Comment:
From my understanding this is just creating a relationship between items that match to use in the graph by taking all of the matching pageIds and offsetting them like so: [1,2,3,4] -> zip([1,2,3,4],[2,3,4,1] = [(1,2),(2,3),(3,4),(4,1)]. Make sense, just needs a comment explaining the logic

Reply:
Yes, that's the logic. Will add the suggested comment.